### PR TITLE
Alter rename column

### PR DESCRIPTION
**Pull request by alesapin**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- New Feature

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Add `ALTER TABLE ... RENAME COLUMN` for MergeTree table engines family.

### CODE REVIEW

**Comment by alesapin** [comment]
Status:
[ ] doesn't work for compact parts.
[ ] doesn't work for nested columns.

**Comment by alesapin** [comment]
It's not ready for review. Code for polymorphic parts left.

File: dbms/Interpreters/InterpreterAlterQuery.cpp [hunk 1]
```diff
@@ -280,6 +280,10 @@ AccessRightsElements InterpreterAlterQuery::getRequiredAccessForCommand(const AS
             required_access.emplace_back(AccessType::REFRESH_VIEW, database, table);
             break;
         }
+        case ASTAlterCommand::RENAME_COLUMN:
```
Comment (alesapin) [inline dbms/Interpreters/InterpreterAlterQuery.cpp:283]: Need to fix this.

File: dbms/Parsers/ParserAlterQuery.cpp [hunk 2]
```diff
@@ -121,6 +123,22 @@ bool ParserAlterCommand::parseImpl(Pos & pos, ASTPtr & node, Expected & expected
 
             command->type = ASTAlterCommand::ADD_COLUMN;
         }
+        else if (s_rename_column.ignore(pos, expected))
+        {
+            if (s_if_exists.ignore(pos, expected))
+                command->if_not_exists = true;
```
Comment (tavplubix) [inline dbms/Parsers/ParserAlterQuery.cpp:129]: Did you mean `command->if_exists`?

Comment (alesapin) [reply dbms/Parsers/ParserAlterQuery.cpp:129]: Yes, thanks.

**Review by tavplubix — APPROVED** [review]

**Comment by simPod** [comment]
@alesapin any idea in which version this will be included? I went with `20.1.11.73` but it's not there yet. Thanks!

**Comment by alex-zaitsev** [comment]
@simPod , this is for 20.4. You can check the tags of merge commits:
https://github.com/ClickHouse/ClickHouse/commit/f1e0f155565ca2601312ade404c8964de231a36d

**Comment by vkingnew** [comment]
it works.
>Clickhouse> alter table t_city rename column city_level TO  cityLevel;

>ALTER TABLE t_city
    RENAME COLUMN city_level TO cityLevel


Ok.

0 rows in set. Elapsed: 0.018 sec.

**Pull request merged**